regression-tests/mixed-type-safety-1.cpp2

```diff
@@ -10,8 +10,9 @@ class Square : public Shape { };
 
 //--- printing helpers -----------------
 
-print: <T : type> ( msg: std::string, x: T ) 
-       requires !std::convertible_to<T, bool> =
+non_bool: <T> concept = !std::convertible_to<T, bool>;
```

Comment:
As mentioned in the description of the PR replacing the `requires` expression using this `concept`  allows GCC 10 to compile the generated C++ code.

Reply:
I didn't know that we could write concepts in CPP2 already. Thank you for showing that.

Reply:
It was a bit of a stab in the dark, but it worked. In fact this overload is not used at all in this test. One of the first things I tried was simply removing it altogether, which also made GCC 10 happy.
Then I tried to come up with a way to rewrite it to be able to keep it.